python/paddle/fluid/dygraph/dygraph_to_static/break_continue_transformer.py

```diff
@@ -0,0 +1,353 @@
+        # TODO: because we change ancestor nodes during visit_Break/Continue,
+        # not current node, so generic_visit of NodeTransformer will visit node
+        # which may be deleted. To prevent that node being added into
+        # transformed AST, I have to self-write a generic_visit, but this is
+        # NOT a good thing. Considering refactorying this whole class.
+        for field, value in gast.iter_fields(node):
+            if isinstance(value, list):
+                for item in value:
+                    if isinstance(item, gast.AST):
+                        self.visit(item)
+            elif isinstance(value, gast.AST):
+                self.visit(value)
+
+    def visit(self, node):
+        self.ancestor_nodes.append(node)
+        method = 'visit_' + node.__class__.__name__
+        visitor = getattr(self, method, self.generic_visit)
+        ret = visitor(node)
+        self.ancestor_nodes.pop()
+        return ret
+
+    def visit_Break(self, node):
+        loop_node_index = self._find_ancestor_loop_index(node)
+        assert loop_node_index != -1, "SyntaxError: 'break' outside loop"
+        loop_node = self.ancestor_nodes[loop_node_index]
+
+        # 1. Map the 'break/continue' stmt with an unique boolean variable V.
+        variable_name = unique_name.generate(BREAK_NAME_PREFIX)
+
+        # 2. Find the first ancestor block containing this 'break/continue', a
+        # block can be a node containing stmt list. We should remove all stmts
+        # after the 'break/continue' and set the V to True here.
+        first_block_index = self._remove_stmts_after_break_continue(
+            node, variable_name, loop_node_index)
+
+        # 3. Add 'if V' for stmts in ancestor blocks between the first one
+        # (exclusive) and the ancestor loop (inclusive)
+        self._replace_if_stmt(loop_node_index, first_block_index, variable_name)
+
+        # 4. For 'break' add break into condition of the loop.
+        assign_false_node = create_fill_constant_node(variable_name, False)
+        self._add_stmt_before_cur_node(loop_node_index, assign_false_node)
+
+        cond_var_node = gast.UnaryOp(
+            op=gast.Not(),
+            operand=gast.Name(
+                id=variable_name,
+                ctx=gast.Load(),
+                annotation=None,
+                type_comment=None))
+        if isinstance(loop_node, gast.While):
+            loop_node.test = gast.BoolOp(
+                op=gast.And(), values=[loop_node.test, cond_var_node])
+        elif isinstance(loop_node, gast.For):
+            parent_node = self.ancestor_nodes[loop_node_index - 1]
+            for_to_while = ForToWhileTransformer(parent_node, loop_node,
+                                                 cond_var_node)
+            for_to_while.transform()
+
+    def visit_Continue(self, node):
+        loop_node_index = self._find_ancestor_loop_index(node)
+        assert loop_node_index != -1, "SyntaxError: 'break' outside loop"
```

Comment:
The message may be "SyntaxError: 'continue' outside loop"

Reply:
Done. These lines were copied so I included the copied error :-)